crates/pixi_api/src/workspace/workspace/environment.rs

```diff
@@ -0,0 +1,77 @@
+use miette::IntoDiagnostic;
+use pixi_core::{
+    Workspace,
+    workspace::{Environment, WorkspaceMut},
+};
+use pixi_manifest::EnvironmentName;
+
+use crate::Interface;
+
+pub async fn list(workspace: &Workspace) -> Vec<Environment> {
+    workspace.environments()
+}
+
+pub async fn add<I: Interface>(
+    interface: &I,
+    mut workspace: WorkspaceMut,
+    name: EnvironmentName,
+    features: Option<Vec<String>>,
+    solve_group: Option<String>,
+    no_default_feature: bool,
+    force: bool,
+) -> miette::Result<()> {
+    let environment_exists = workspace.workspace().environment(&name).is_some();
+    if environment_exists && !force {
+        return Err(miette::miette!(
+            help = "use --force to overwrite the existing environment",
+            "the environment '{}' already exists",
```

Comment:
That error message is only applicable for the CLI, but not e.g. for the GUI. I think the CLI should add this context